jlm/util/Hash.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Nico Reißmann <nico.reissmann@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_UTIL_HASH_HPP
+#define JLM_UTIL_HASH_HPP
+
+#include <functional>
+#include <string_view>
+
+namespace jlm::util
+{
+
+/**
+ * Our own version of std::hash that also supports hashing std::pair
+ */
+template<typename T>
+struct Hash : std::hash<T>
+{
+};
+
+template<typename First, typename Second>
+struct Hash<std::pair<First, Second>>
+{
+  std::size_t
+  operator()(const std::pair<First, Second> & value) const noexcept
+  {
+    return std::hash<First>()(value.first) ^ std::hash<Second>()(value.second) << 1;
```

Comment:
If you want to you can replace `std::hash` with `Hash` to enable hashing pairs of pairs as well.